libraries/AP_GPS/AP_GPS.cpp

```diff
@@ -562,10 +545,10 @@ void AP_GPS::detect_instance(uint8_t instance)
     }
 
 found_gps:
-    if (new_gps != nullptr) {
+    if (new_gps != nullptr && instance < GPS_MAX_RECEIVERS) {
```

Comment:
If `new_gps != nullptr` but `instance >= GPS_MAX_RECEIVERS` then the `new_gps` should get destructed.

Reply:
This is an extraneous check, code would never get here with instance >= GPS_MAX_RECEIVERS

Reply:
If you want to leave the `instance >= GPS_MAX_RECEIVERS` check, just move in to the beginning with `return`.